frontend/integration-tests/tests/secrets.scenario.ts

```diff
@@ -1,64 +1,119 @@
+      await secretsView.checkSecret(SECRET_NAMES.webhook, [secretsView.webhookSecretValue]);
     });
 
     it('edits webhook secret', async() => {
-      await crudView.actionsDropdown.click();
-      await browser.wait(until.presenceOf(crudView.actionsDropdownMenu), 500);
-      await crudView.actionsDropdownMenu.element(by.linkText('Edit Secret')).click();
-      await browser.wait(until.urlContains(`/k8s/ns/${testName}/secrets/${SECRET_NAME}/edit`));
-      await element(by.buttonText('Generate')).click();
-      await secretsView.saveButton.click();
-      expect(secretsView.errorMessage.isPresent()).toBe(false);
+      await secretsView.editSecret(testName, SECRET_NAMES.webhook, async() => {
+        await element(by.buttonText('Generate')).click();
+      });
     });
 
     it('check for edited webhook secret value', async() => {
-      await browser.wait(until.textToBePresentInElement($('.co-m-pane__heading'), SECRET_NAME));
+      await browser.wait(until.textToBePresentInElement($('.co-m-pane__heading'), SECRET_NAMES.webhook));
       await element(by.partialButtonText('Reveal Values')).click();
       expect(secretsView.pre.get(0).getText()).not.toEqual(secretsView.webhookSecretValue);
     });
   });
+
+  describe('Basic source secrets', () => {
+    sharedSteps();
+
+    it('creates basic source secret', async() => {
+      await secretsView.createSecret(secretsView.createSourceSecretLink, SECRET_NAMES.basicSource, async() => {
+        await secretsView.sourceSecretUsernameInput.sendKeys(secretsView.sourceSecretUsername);
+        await secretsView.sourceSecretPasswordInput.sendKeys(secretsView.sourceSecretPassword);
+      });
+    });
+
+    it('check for created basic source secret values', async() => {
+      await secretsView.checkSecret(SECRET_NAMES.basicSource, [secretsView.sourceSecretPassword, secretsView.sourceSecretUsername]);
+    });
+
+    it('edits basic source secret', async() => {
+      await secretsView.editSecret(testName, SECRET_NAMES.basicSource, async() => {
+        await secretsView.sourceSecretUsernameInput.clear();
+        await secretsView.sourceSecretUsernameInput.sendKeys(secretsView.sourceSecretUsernameUpdated);
+        await secretsView.sourceSecretPasswordInput.clear();
+        await secretsView.sourceSecretPasswordInput.sendKeys(secretsView.sourceSecretPasswordUpdated);
+      });
+    });
+
+    it('check for edited basic source secret values', async() => {
+      await secretsView.checkSecret(SECRET_NAMES.basicSource, [secretsView.sourceSecretPasswordUpdated, secretsView.sourceSecretUsernameUpdated]);
+    });
+  });
+
+  describe('SSH source secrets', () => {
+    sharedSteps();
+
+    it('creates SSH source secret', async() => {
+      await secretsView.createSecret(secretsView.createSourceSecretLink, SECRET_NAMES.sshSource, async() => {
+        await secretsView.authTypeDropdown.click().then(() => browser.actions().sendKeys(Key.ARROW_UP, Key.ENTER).perform());
```

Comment:
Had to add `/*eslint max-nested-callbacks: ["error", 5]*/` because of this step, since it's already nested and also is wrapped in the function.